pkg/test/vpc_client/bastion.go

```diff
@@ -94,55 +96,45 @@ func (vpc *VPC) LaunchBastion(imageID string, zone string, userData string, keyp
 	return inst, nil
 }
 
-func (vpc *VPC) PrepareBastionProxy(zone string, cidrBlock string, keypairName string,
-	privateKeyPath string) (*types.Instance, error) {
-	filters := []map[string][]string{
-		{
-			"vpc-id": {
-				vpc.VpcID,
-			},
-		},
-		{
-			"tag:Name": {
-				CON.BastionName,
-			},
-		},
-	}
-
-	insts, err := vpc.AWSClient.ListInstances([]string{}, filters...)
+// PrepareBastionProxy will launch a bastion instance with squid proxy on the indicated zone and return the proxy url.
+func (vpc *VPC) PrepareBastionProxy(zone string, keypairName string, privateKeyPath string) (proxyUrl string, err error) {
+	userData := generateShellCommand()
+
+	encodeUserData := base64.StdEncoding.EncodeToString([]byte(userData))
+	instance, err := vpc.LaunchBastion("", zone, encodeUserData, keypairName, privateKeyPath)
 	if err != nil {
-		return nil, err
+		log.LogError("Launch bastion failed")
 	}
-	if len(insts) == 0 {
-		log.LogInfo("Didn't found an existing bastion, going to launch one")
-		if cidrBlock == "" {
-			cidrBlock = CON.RouteDestinationCidrBlock
-		}
-		_, _, err = net.ParseCIDR(cidrBlock)
-		if err != nil {
-			log.LogError("CIDR IP address format is invalid")
-			return nil, err
-		}
 
-		userData := fmt.Sprintf(`#!/bin/bash
-		yum update -y
-		yum install -y squid
-		cd /etc/squid/
-		sudo mv ./squid.conf ./squid.conf.bak
-		sudo touch squid.conf
-		echo http_port 3128 >> /etc/squid/squid.conf
-		echo acl allowed_ips src %s >> /etc/squid/squid.conf
-		echo http_access allow allowed_ips >> /etc/squid/squid.conf
-		echo http_access deny all >> /etc/squid/squid.conf
-		systemctl start squid
-		systemctl enable squid`, cidrBlock)
+	username := utils.RandomLabel(5)
+	password := utils.RandomLabel(10)
+
+	hashedPassword, err := generateBcryptPassword(password)
+	if err != nil {
+		return "", err
+	}
 
-		encodeUserData := base64.StdEncoding.EncodeToString([]byte(userData))
-		return vpc.LaunchBastion("", zone, encodeUserData, keypairName, privateKeyPath)
+	line := fmt.Sprintf("%s:%s\n", username, hashedPassword)
+	remoteFilePath := "/etc/squid/passwords"
+	hostName := fmt.Sprintf("%s:22", *instance.PublicIpAddress)
+	createFileCMD := fmt.Sprintf("sudo touch %s", remoteFilePath)
+	copyPasswordCMD := fmt.Sprintf("echo '%s' | sudo tee %s > /dev/null", line, remoteFilePath)
+	SSHExecuteCMDs := []string{
+		createFileCMD,
+		copyPasswordCMD,
+	}
+	privateKeyName := fmt.Sprintf("%s-%s", path.Join(privateKeyPath, keypairName), "keyPair.pem")
```

Comment:
Make the keypath to a common function. Otherwise I have to maintain too many places and it will cause unexpected error during running.

Reply:
Updated, create a new func GetPrivateKeyName() in aws utils.